### PR TITLE
[MAINTENANCE] Call the correct sqlalchemy method to eliminate deprecation warning

### LINKED ISSUE #5724: Deprecation warnings when calling Great Expectations data context (connecting to S3)
**Describe the bug**
I'm getting a deprecation warning when I call data context, specifically when Great Expectations is configured to pulled expectation suites from S3. 

The warning seems to be an [SQLALchemy deprecation warning](https://github.com/zzzeek/sqlalchemy/blob/master/lib/sqlalchemy/engine/url.py):
```
venv-validation/lib/python3.9/site-packages/great_expectations/data_context/store/query_store.py:73: SADeprecationWarning: Calling URL() directly is deprecated and will be disabled in a future release.  The public constructor for URL is now the URL.create() method options = URL(drivername, **credentials)
```

**To Reproduce**
1. Configure great_expectations.yml with S3 expectation store
2. Create a data context object: `context = ge.data_context.DataContext()`
3. See error

**Expected behavior**
The data context does not show the warnings when creating a data context connecting to local expectation suites. 

**Environment (please complete the following information):**
 - Operating System: MacOS
 - Great Expectations Version: 0.15.16

### PR DESCRIPTION
**Pull request by anthonyburdi**
Changes proposed in this pull request:
- Call a different sqlalchemy method based on version to eliminate a deprecation warning.
- Closes https://github.com/great-expectations/great_expectations/issues/5724

### Definition of Done

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added [unit tests](https://docs.greatexpectations.io/docs/contributing/contributing_test#writing-unit-and-integration-tests) where applicable and made sure that new and existing tests are passing.
- [x] I have run any local integration tests and made sure that nothing is broken.

### CODE REVIEW

File: tests/data_context/store/test_query_store.py [hunk 1]
```diff
@@ -14,7 +14,7 @@ def basic_sqlalchemy_query_store(titanic_sqlite_db):
 
 
 @pytest.fixture()
-def test_basic_sqlalchemy_query_store_connection_string(
+def basic_sqlalchemy_query_store_connection_string(
```
Comment (anthonyburdi) [inline tests/data_context/store/test_query_store.py:20]: Unrelated change, just saw an issue in the same file. Fixtures shouldn't start with `test_`

Comment (anthonyburdi) [reply tests/data_context/store/test_query_store.py:20]: (just trying to keep the campsite cleaner than I found it 😄)

File: great_expectations/data_context/store/query_store.py [hunk 2]
```diff
@@ -4,16 +4,23 @@
...
+from great_expectations.optional_imports import is_version_greater_or_equal
 from great_expectations.util import filter_properties_dict
 
 try:
     import sqlalchemy
     from sqlalchemy import create_engine
     from sqlalchemy.engine.url import URL
+
+    if is_version_greater_or_equal(sqlalchemy.__version__, "1.4.0"):
+        url_create_fn = URL.create
```
Comment (Shinnnyshinshin) [inline great_expectations/data_context/store/query_store.py:16]: :heart:

File: great_expectations/optional_imports.py [hunk 3]
```diff
@@ -28,3 +28,23 @@ def sqlalchemy_version_check(version: str | Version) -> None:
             "SQLAlchemy v2.0.0 or later is not yet supported by Great Expectations.",
             UserWarning,
         )
+
+
+def is_version_greater_or_equal(
```
Comment (Shinnnyshinshin) [inline great_expectations/optional_imports.py:33]: Again, love the approach and solution. Thank you :)

**Review by Shinnnyshinshin — APPROVED** [review]
Love it. Thank you @anthonyburdi
